Tests/AK/TestWeakPtr.cpp

```diff
@@ -15,8 +15,8 @@
 #    pragma clang diagnostic ignored "-Wunused-private-field"
 #endif
 
-class SimpleWeakable : public Weakable<SimpleWeakable>
-    , public RefCounted<SimpleWeakable> {
+class SimpleWeakable : public RefCounted<SimpleWeakable>
```

Comment:
As I mentioned in the comments of sc7's PR (with test cases to prove it!), this is *only* an assumption of LockRefPtr. RefPtr can handle these cases just fine. I don't think it's worth the effort to enforce this for things that are not held in LockRefPtr. And since we intend to remove LockRefPtr as soon as enough folks get the motivation to use proper locking for things in the kernel, I'm not sure that it's a problem we need to solve in general.

Reply:
Right, and with #17847 we can technically just treat it as "problem solved".